macros/src/parser/mod.rs

```diff
@@ -27,6 +27,9 @@ pub struct AsyncIdent {
 
 #[derive(Debug)]
 pub struct ParsedStateMachine {
+    pub name: Option<Ident>,
+    pub derive_states: Vec<Ident>,
+    pub derive_events: Vec<Ident>,
     pub temporary_context_type: Option<Type>,
     pub custom_guard_error: bool,
     pub impl_display_events: bool,
```

Comment:
Hmm, tbh now that we have the custom derive keywords, it seems like we should remove the `impl_display_XXX` flags and make it so there's a single way to add derives to the states and events